templates/ruby/lib/container/payload.rb.twig

```diff
@@ -0,0 +1,81 @@
+module Appwrite
+  class Payload
+    attr_accessor :data, :mime_type, :filename, :path
+
+    def size 
+      unless @data.nil?
+        return @data.bytesize
+      end
+      unless @path.nil?
+        return File.size(@path)
+      end
+      raise "Empty payload"
+    end
+
+    def initialize(data = nil, path = nil, filename = nil, mime_type = nil)
+      @path = path
+      @data = data
+
+      @filename = if filename.nil? and path then
+        File.basename(path)
+      else
+        filename
+      end
+
+      @mime_type = if mime_type.nil? and path then
+        MIME::Types.type_for(path).first.content_type rescue nil
+      else
+        mime_type
+      end
+    end
+
+    def self.from_path(path, filename: nil, mime_type: nil)
+      raise "File not found" unless File.exist?(path)
+      new(nil, path, filename, mime_type)
+    end
+
+    def self.from_data(data, filename: nil, mime_type: nil)
+      new(data, nil, filename, mime_type)
+    end
+
+    def self.from_json(data, filename: nil, mime_type: 'application/json')
+      new(JSON.generate(data), nil, filename, mime_type)
+    end
+
+    def self.from_string(data, filename: nil, mime_type: 'text/plain')
+      new(data, nil, filename, mime_type)
+    end
+
+    def to_binary
+      @data || File.binread(@path)
+    end
+
+    def to_json
+      JSON.parse(@data)
+    end
+
+    def to_s
```

Comment:
is it not possible to define a to_string? it it a reserved function ?

Reply:
to_string is not reserved, but to_s is idiomatic 

I've gone with both